login/service.go

```diff
@@ -0,0 +1,118 @@
+package login
+
+// service.go contains the definition and implementation (business logic) of the
+// login service. Everything here is agnostic to the transport (HTTP).
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strings"
+)
+
+// Service is the login service, providing operations for users to login and register.
+type Service interface {
+	Login(username, password string) (User, error)   // GET /login
+	// Only used for testing at the moment
+	Register(username, password string) bool	// GET /register?username=[username]&password=[password]
+}
+
+// User describes the logged in user.
+type User struct {
+	Id       string `json:"id"`
+	Name     string `json:"name"`
+	Password string `json:"password"`
+	Link     string `json:"link"`
+}
+
+// ErrNotAuthorized is returned when there the supplied credentials are not valid.
+var ErrNotAuthorized = errors.New("invalid credendtials")
+
+// For Customer Lookup
+var customerHost = "accounts"
+var customerLookupPath = "/customers/search/findByUsername"
+
+// NewFixedService returns a simple implementation of the Service interface,
+// fixed over a predefined set of users. Replace with db integration?
+func NewFixedService(users []User) Service {
+	return &fixedService{
+		users: users,
+	}
+}
+
+type fixedService struct {
+	users []User
+}
+
+func (s *fixedService) Login(username, password string) (User, error) {
+	found := false
+	for _, user := range s.users {
+		if user.Name == username && user.Password == password {
+			found = true
+		}
+	}
+
+	if !found {
+		return User{}, ErrNotAuthorized
```

Comment:
Ah! I understand now, this method is called when the user is trying to log in, and Unauthorized means both "Wrong password" and "No such user found" and "Other general classes of failure". LGTM.